tests/test_routes.py

```diff
@@ -147,14 +148,63 @@ def test_not_found(self):
         self.assertEqual(self.client.get("/not-found-url").status_code, 404)
 
     def test_engage_index(self):
-        """
-        When given a /engage URL,
-        we should return a 200 status code
-        This also tests that discourse module
-        for engage pages work
-        """
+        response = self.client.get("/engage")
+        self.assertEqual(response.status_code, 200)
+
+    def test_engage_index_without_preview_flag_cannot_see_inactive_page(self):
+        response = self.client.get("/engage")
+        self.assertEqual(response.status_code, 200)
+
+        soup = BeautifulSoup(response.data, "html.parser")
+        self.assertIsNone(
+            soup.find("a", {"href": "/engage/it/deployment-azienda-manuale"})
```

Comment:
Is it likely that this post will become active at some point?

Reply:
I will make sure to make it active before this gets merged in. 

But the test will be still valid due to how the cassettes work. They test against a cached `/engage` where the post will always inactive.